compiler.js

```diff
@@ -1,3 +1,68 @@
+let examples = new Map();
+
+examples["helloWorld"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+contract MyContract {
+    function helloWorld() public pure returns (string memory) {
+        return "Hello, World!";
+    }
+}`;
+
+examples["erc20"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ERC20 {
+    string public constant name = "ERC20";
+    string public constant symbol = "ERC";
+    uint8 public constant decimals = 18;
+
+    mapping(address => uint256) balances;
+    mapping(address => mapping (address => uint256)) allowed;
+
+    event Transfer(address indexed from, address indexed to, uint256 tokens);
+    event Approval(address indexed owner, address indexed spender, uint256 tokens);
+    
+    uint256 m_totalSupply;
+    
+    constructor(uint256 total) {
+        m_totalSupply = total;
+        balances[msg.sender] = m_totalSupply;
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return m_totalSupply;
+    }
+    
+    function balanceOf(address tokenOwner) public view returns (uint256) {
+        return balances[tokenOwner];
+    }
+    
+    function transfer(address receiver, uint256 numTokens) public returns (bool) {
+        balances[msg.sender] = balances[msg.sender] - numTokens;
```

Comment:
Should we add a `require` here to avoid the panic?

Reply:
yes, added both here and in  transferFrom funcion